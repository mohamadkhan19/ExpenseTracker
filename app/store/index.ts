import { configureStore } from '@reduxjs/toolkit';
import { baseApi } from './api/baseApi';
import expensesReducer from './slices/expenses.slice';

export const store = configureStore({
  reducer: {
    expenses: expensesReducer,
    [baseApi.reducerPath]: baseApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: ['persist/PERSIST', 'persist/REHYDRATE'],
      },
    }).concat(baseApi.middleware),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
