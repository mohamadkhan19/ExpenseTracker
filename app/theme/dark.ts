import { darkColors } from './tokens/colors';
import { spacing } from './tokens/spacing';
import { fontSizes, fontWeights } from './tokens/typography';
import { radii } from './tokens/radii';
import { zIndex } from './tokens/zIndex';

export const darkTheme = {
  colors: darkColors,
  spacing,
  fontSizes,
  fontWeights,
  radii,
  zIndex,
  mode: 'dark' as const,
};


