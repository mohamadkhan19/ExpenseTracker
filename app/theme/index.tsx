import React, { createContext, useContext, useMemo, useState, useEffect } from 'react';
import { useColorScheme } from 'react-native';
import { lightTheme, AppTheme } from './light';
import { darkTheme } from './dark';
import { AsyncStorageClient } from '../services/storage/asyncStorageClient';
import { STORAGE_KEYS } from '../services/storage/keys';
import { logger } from '../utils/logger';

type ThemeMode = 'light' | 'dark' | 'system';

interface ThemeContextType {
  theme: AppTheme;
  themeMode: ThemeMode;
  setThemeMode: (mode: ThemeMode) => void;
  toggleTheme: () => void;
}

const ThemeContext = createContext<ThemeContextType | undefined>(undefined);

export function ThemeProvider({ children }: { children: React.ReactNode }) {
  const systemColorScheme = useColorScheme();
  const [themeMode, setThemeModeState] = useState<ThemeMode>('system');

  // Load saved theme preference on mount
  useEffect(() => {
    const loadThemePreference = async () => {
      try {
        const savedTheme = await AsyncStorageClient.get<ThemeMode>(STORAGE_KEYS.THEME);
        if (savedTheme && ['light', 'dark', 'system'].includes(savedTheme)) {
          setThemeModeState(savedTheme);
        }
      } catch (error) {
        logger.error('Failed to load theme preference', error as Error);
      }
    };
    loadThemePreference();
  }, []);

  // Save theme preference when it changes
  const setThemeMode = async (mode: ThemeMode) => {
    setThemeModeState(mode);
    try {
      await AsyncStorageClient.set(STORAGE_KEYS.THEME, mode);
    } catch (error) {
      logger.error('Failed to save theme preference', error as Error, { mode });
    }
  };

  // Toggle between light and dark (ignoring system)
  const toggleTheme = () => {
    const newMode = themeMode === 'light' ? 'dark' : 'light';
    setThemeMode(newMode);
  };

  // Determine which theme to use
  const theme = useMemo(() => {
    if (themeMode === 'system') {
      return systemColorScheme === 'dark' ? darkTheme : lightTheme;
    } else {
      return themeMode === 'dark' ? darkTheme : lightTheme;
    }
  }, [themeMode, systemColorScheme]);

  const contextValue = useMemo(() => ({
    theme,
    themeMode,
    setThemeMode,
    toggleTheme,
  }), [theme, themeMode]);

  return <ThemeContext.Provider value={contextValue}>{children}</ThemeContext.Provider>;
}

export function useTheme() {
  const context = useContext(ThemeContext);
  if (context === undefined) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }
  return context;
}


