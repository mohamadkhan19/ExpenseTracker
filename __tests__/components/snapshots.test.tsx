import React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from '../../app/theme';
import { ExpenseCard } from '../../app/ui/organisms/ExpenseCard';
import { Button } from '../../app/ui/atoms/Button';
import { Input } from '../../app/ui/atoms/Input';

// Mock the logger to avoid console output during tests
jest.mock('../../app/utils/logger', () => ({
  logger: {
    debug: jest.fn(),
    info: jest.fn(),
    warn: jest.fn(),
    error: jest.fn(),
  },
}));

// Test wrapper component
const TestWrapper = ({ children }: { children: React.ReactNode }) => (
  <ThemeProvider>{children}</ThemeProvider>
);

describe('UI Component Snapshot Tests', () => {
  describe('ExpenseCard', () => {
    const mockExpense = {
      id: '1',
      description: 'Coffee',
      amount: 4.50,
      category: 'Food',
      date: '2023-01-01',
    };

    it('should match snapshot for basic expense card', () => {
      const tree = render(
        <TestWrapper>
          <ExpenseCard expense={mockExpense} onPress={jest.fn()} />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot for expense card with long description', () => {
      const longDescriptionExpense = {
        ...mockExpense,
        description: 'This is a very long description that should test text wrapping and layout',
      };

      const tree = render(
        <TestWrapper>
          <ExpenseCard expense={longDescriptionExpense} onPress={jest.fn()} />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot for expense card with large amount', () => {
      const largeAmountExpense = {
        ...mockExpense,
        amount: 1234.56,
      };

      const tree = render(
        <TestWrapper>
          <ExpenseCard expense={largeAmountExpense} onPress={jest.fn()} />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot for expense card with onLongPress', () => {
      const tree = render(
        <TestWrapper>
          <ExpenseCard 
            expense={mockExpense} 
            onPress={jest.fn()} 
            onLongPress={jest.fn()} 
          />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Button', () => {
    it('should match snapshot for primary button', () => {
      const tree = render(
        <TestWrapper>
          <Button title="Primary Button" onPress={jest.fn()} />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot for secondary button', () => {
      const tree = render(
        <TestWrapper>
          <Button title="Secondary Button" variant="secondary" onPress={jest.fn()} />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot for disabled button', () => {
      const tree = render(
        <TestWrapper>
          <Button title="Disabled Button" disabled onPress={jest.fn()} />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot for button with custom style', () => {
      const tree = render(
        <TestWrapper>
          <Button 
            title="Custom Button" 
            onPress={jest.fn()}
            style={{ backgroundColor: 'red' }}
          />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot for small button', () => {
      const tree = render(
        <TestWrapper>
          <Button title="Small Button" size="small" onPress={jest.fn()} />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot for large button', () => {
      const tree = render(
        <TestWrapper>
          <Button title="Large Button" size="large" onPress={jest.fn()} />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Input', () => {
    it('should match snapshot for basic input', () => {
      const tree = render(
        <TestWrapper>
          <Input 
            placeholder="Enter text" 
            value="" 
            onChangeText={jest.fn()} 
          />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot for input with value', () => {
      const tree = render(
        <TestWrapper>
          <Input 
            placeholder="Enter text" 
            value="Some text" 
            onChangeText={jest.fn()} 
          />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot for input with error', () => {
      const tree = render(
        <TestWrapper>
          <Input 
            placeholder="Enter text" 
            value="" 
            onChangeText={jest.fn()}
            error="This field is required"
          />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot for disabled input', () => {
      const tree = render(
        <TestWrapper>
          <Input 
            placeholder="Enter text" 
            value="" 
            onChangeText={jest.fn()}
            disabled
          />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot for multiline input', () => {
      const tree = render(
        <TestWrapper>
          <Input 
            placeholder="Enter description" 
            value="" 
            onChangeText={jest.fn()}
            multiline
            numberOfLines={3}
          />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });

    it('should match snapshot for numeric input', () => {
      const tree = render(
        <TestWrapper>
          <Input 
            placeholder="Enter amount" 
            value="" 
            onChangeText={jest.fn()}
            keyboardType="numeric"
          />
        </TestWrapper>
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
