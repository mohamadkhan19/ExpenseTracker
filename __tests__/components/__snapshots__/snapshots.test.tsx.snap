// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Component Snapshot Tests Button should match snapshot for button with custom style 1`] = `
<View
  accessibilityHint="Tap to custom button"
  accessibilityLabel="Custom Button"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "red",
      "borderRadius": 6,
      "elevation": 1,
      "justifyContent": "center",
      "minHeight": 44,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
      "shadowColor": "#1A1A1A",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.05,
      "shadowRadius": 1,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 15,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    Custom Button
  </Text>
</View>
`;

exports[`UI Component Snapshot Tests Button should match snapshot for disabled button 1`] = `
<View
  accessibilityHint="Tap to disabled button"
  accessibilityLabel="Disabled Button"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#00C805",
      "borderRadius": 6,
      "elevation": 1,
      "justifyContent": "center",
      "minHeight": 44,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
      "shadowColor": "#1A1A1A",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.05,
      "shadowRadius": 1,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 15,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    Disabled Button
  </Text>
</View>
`;

exports[`UI Component Snapshot Tests Button should match snapshot for large button 1`] = `
<View
  accessibilityHint="Tap to large button"
  accessibilityLabel="Large Button"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#00C805",
      "borderRadius": 6,
      "elevation": 1,
      "justifyContent": "center",
      "minHeight": 44,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
      "shadowColor": "#1A1A1A",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.05,
      "shadowRadius": 1,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 15,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    Large Button
  </Text>
</View>
`;

exports[`UI Component Snapshot Tests Button should match snapshot for primary button 1`] = `
<View
  accessibilityHint="Tap to primary button"
  accessibilityLabel="Primary Button"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#00C805",
      "borderRadius": 6,
      "elevation": 1,
      "justifyContent": "center",
      "minHeight": 44,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
      "shadowColor": "#1A1A1A",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.05,
      "shadowRadius": 1,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 15,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    Primary Button
  </Text>
</View>
`;

exports[`UI Component Snapshot Tests Button should match snapshot for secondary button 1`] = `
<View
  accessibilityHint="Tap to secondary button"
  accessibilityLabel="Secondary Button"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F8F9FA",
      "borderRadius": 6,
      "elevation": 1,
      "justifyContent": "center",
      "minHeight": 44,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
      "shadowColor": "#1A1A1A",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.05,
      "shadowRadius": 1,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      [
        {
          "color": "#1A1A1A",
          "fontSize": 15,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    Secondary Button
  </Text>
</View>
`;

exports[`UI Component Snapshot Tests Button should match snapshot for small button 1`] = `
<View
  accessibilityHint="Tap to small button"
  accessibilityLabel="Small Button"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#00C805",
      "borderRadius": 6,
      "elevation": 1,
      "justifyContent": "center",
      "minHeight": 44,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
      "shadowColor": "#1A1A1A",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.05,
      "shadowRadius": 1,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 15,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    Small Button
  </Text>
</View>
`;

exports[`UI Component Snapshot Tests ExpenseCard should match snapshot for basic expense card 1`] = `
<View
  accessibilityHint="Tap to edit expense"
  accessibilityLabel="Expense: Coffee, $4.50, Food, Dec 31, 2022"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#E5E7EB",
      "borderRadius": 6,
      "borderWidth": 1,
      "elevation": 1,
      "marginBottom": 8,
      "opacity": 1,
      "padding": 12,
      "shadowColor": "#1A1A1A",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.05,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 17,
              "fontWeight": "600",
            },
            {
              "color": "#EF4444",
            },
          ]
        }
      >
        -
        $4.50
      </Text>
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#6B7280",
              "fontSize": 13,
              "fontWeight": "400",
            },
            {
              "marginTop": 4,
            },
          ]
        }
      >
        Coffee
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "flex-end",
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#6B7280",
              "fontSize": 13,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        Food
      </Text>
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#6B7280",
              "fontSize": 11,
              "fontWeight": "400",
            },
            {
              "marginTop": 4,
            },
          ]
        }
      >
        Dec 31, 2022
      </Text>
    </View>
  </View>
</View>
`;

exports[`UI Component Snapshot Tests ExpenseCard should match snapshot for expense card with large amount 1`] = `
<View
  accessibilityHint="Tap to edit expense"
  accessibilityLabel="Expense: Coffee, $1234.56, Food, Dec 31, 2022"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#E5E7EB",
      "borderRadius": 6,
      "borderWidth": 1,
      "elevation": 1,
      "marginBottom": 8,
      "opacity": 1,
      "padding": 12,
      "shadowColor": "#1A1A1A",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.05,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 17,
              "fontWeight": "600",
            },
            {
              "color": "#EF4444",
            },
          ]
        }
      >
        -
        $1234.56
      </Text>
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#6B7280",
              "fontSize": 13,
              "fontWeight": "400",
            },
            {
              "marginTop": 4,
            },
          ]
        }
      >
        Coffee
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "flex-end",
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#6B7280",
              "fontSize": 13,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        Food
      </Text>
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#6B7280",
              "fontSize": 11,
              "fontWeight": "400",
            },
            {
              "marginTop": 4,
            },
          ]
        }
      >
        Dec 31, 2022
      </Text>
    </View>
  </View>
</View>
`;

exports[`UI Component Snapshot Tests ExpenseCard should match snapshot for expense card with long description 1`] = `
<View
  accessibilityHint="Tap to edit expense"
  accessibilityLabel="Expense: This is a very long description that should test text wrapping and layout, $4.50, Food, Dec 31, 2022"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#E5E7EB",
      "borderRadius": 6,
      "borderWidth": 1,
      "elevation": 1,
      "marginBottom": 8,
      "opacity": 1,
      "padding": 12,
      "shadowColor": "#1A1A1A",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.05,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 17,
              "fontWeight": "600",
            },
            {
              "color": "#EF4444",
            },
          ]
        }
      >
        -
        $4.50
      </Text>
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#6B7280",
              "fontSize": 13,
              "fontWeight": "400",
            },
            {
              "marginTop": 4,
            },
          ]
        }
      >
        This is a very long description that should test text wrapping and layout
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "flex-end",
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#6B7280",
              "fontSize": 13,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        Food
      </Text>
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#6B7280",
              "fontSize": 11,
              "fontWeight": "400",
            },
            {
              "marginTop": 4,
            },
          ]
        }
      >
        Dec 31, 2022
      </Text>
    </View>
  </View>
</View>
`;

exports[`UI Component Snapshot Tests ExpenseCard should match snapshot for expense card with onLongPress 1`] = `
<View
  accessibilityHint="Tap to edit expense"
  accessibilityLabel="Expense: Coffee, $4.50, Food, Dec 31, 2022"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#E5E7EB",
      "borderRadius": 6,
      "borderWidth": 1,
      "elevation": 1,
      "marginBottom": 8,
      "opacity": 1,
      "padding": 12,
      "shadowColor": "#1A1A1A",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.05,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#1A1A1A",
              "fontSize": 17,
              "fontWeight": "600",
            },
            {
              "color": "#EF4444",
            },
          ]
        }
      >
        -
        $4.50
      </Text>
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#6B7280",
              "fontSize": 13,
              "fontWeight": "400",
            },
            {
              "marginTop": 4,
            },
          ]
        }
      >
        Coffee
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "flex-end",
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#6B7280",
              "fontSize": 13,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        Food
      </Text>
      <Text
        accessibilityRole="text"
        style={
          [
            {
              "color": "#6B7280",
              "fontSize": 11,
              "fontWeight": "400",
            },
            {
              "marginTop": 4,
            },
          ]
        }
      >
        Dec 31, 2022
      </Text>
    </View>
  </View>
</View>
`;

exports[`UI Component Snapshot Tests Input should match snapshot for basic input 1`] = `
<View>
  <TextInput
    accessibilityHint="Enter text"
    accessibilityLabel="Text input"
    accessibilityRole="text"
    onChangeText={[MockFunction]}
    placeholder="Enter text"
    placeholderTextColor="#6B7280"
    style={
      [
        {
          "backgroundColor": "#F8F9FA",
          "borderColor": "#E5E7EB",
          "borderRadius": 6,
          "borderWidth": 1,
          "color": "#1A1A1A",
          "fontSize": 15,
          "minHeight": 44,
          "paddingHorizontal": 12,
          "paddingVertical": 8,
        },
        undefined,
      ]
    }
    value=""
  />
</View>
`;

exports[`UI Component Snapshot Tests Input should match snapshot for disabled input 1`] = `
<View>
  <TextInput
    accessibilityHint="Enter text"
    accessibilityLabel="Text input"
    accessibilityRole="text"
    disabled={true}
    onChangeText={[MockFunction]}
    placeholder="Enter text"
    placeholderTextColor="#6B7280"
    style={
      [
        {
          "backgroundColor": "#F8F9FA",
          "borderColor": "#E5E7EB",
          "borderRadius": 6,
          "borderWidth": 1,
          "color": "#1A1A1A",
          "fontSize": 15,
          "minHeight": 44,
          "paddingHorizontal": 12,
          "paddingVertical": 8,
        },
        undefined,
      ]
    }
    value=""
  />
</View>
`;

exports[`UI Component Snapshot Tests Input should match snapshot for input with error 1`] = `
<View>
  <TextInput
    accessibilityHint="Error: This field is required"
    accessibilityLabel="Text input"
    accessibilityRole="text"
    onChangeText={[MockFunction]}
    placeholder="Enter text"
    placeholderTextColor="#6B7280"
    style={
      [
        {
          "backgroundColor": "#F8F9FA",
          "borderColor": "#EF4444",
          "borderRadius": 6,
          "borderWidth": 1,
          "color": "#1A1A1A",
          "fontSize": 15,
          "minHeight": 44,
          "paddingHorizontal": 12,
          "paddingVertical": 8,
        },
        undefined,
      ]
    }
    value=""
  />
  <Text
    accessibilityRole="text"
    style={
      [
        {
          "color": "#EF4444",
          "fontSize": 11,
          "fontWeight": "400",
        },
        {
          "marginTop": 4,
        },
      ]
    }
  >
    This field is required
  </Text>
</View>
`;

exports[`UI Component Snapshot Tests Input should match snapshot for input with value 1`] = `
<View>
  <TextInput
    accessibilityHint="Enter text"
    accessibilityLabel="Text input"
    accessibilityRole="text"
    onChangeText={[MockFunction]}
    placeholder="Enter text"
    placeholderTextColor="#6B7280"
    style={
      [
        {
          "backgroundColor": "#F8F9FA",
          "borderColor": "#E5E7EB",
          "borderRadius": 6,
          "borderWidth": 1,
          "color": "#1A1A1A",
          "fontSize": 15,
          "minHeight": 44,
          "paddingHorizontal": 12,
          "paddingVertical": 8,
        },
        undefined,
      ]
    }
    value="Some text"
  />
</View>
`;

exports[`UI Component Snapshot Tests Input should match snapshot for multiline input 1`] = `
<View>
  <TextInput
    accessibilityHint="Enter text"
    accessibilityLabel="Text input"
    accessibilityRole="text"
    multiline={true}
    numberOfLines={3}
    onChangeText={[MockFunction]}
    placeholder="Enter description"
    placeholderTextColor="#6B7280"
    style={
      [
        {
          "backgroundColor": "#F8F9FA",
          "borderColor": "#E5E7EB",
          "borderRadius": 6,
          "borderWidth": 1,
          "color": "#1A1A1A",
          "fontSize": 15,
          "minHeight": 44,
          "paddingHorizontal": 12,
          "paddingVertical": 8,
        },
        undefined,
      ]
    }
    value=""
  />
</View>
`;

exports[`UI Component Snapshot Tests Input should match snapshot for numeric input 1`] = `
<View>
  <TextInput
    accessibilityHint="Enter text"
    accessibilityLabel="Text input"
    accessibilityRole="text"
    keyboardType="numeric"
    onChangeText={[MockFunction]}
    placeholder="Enter amount"
    placeholderTextColor="#6B7280"
    style={
      [
        {
          "backgroundColor": "#F8F9FA",
          "borderColor": "#E5E7EB",
          "borderRadius": 6,
          "borderWidth": 1,
          "color": "#1A1A1A",
          "fontSize": 15,
          "minHeight": 44,
          "paddingHorizontal": 12,
          "paddingVertical": 8,
        },
        undefined,
      ]
    }
    value=""
  />
</View>
`;
